File: ModelCabinet.Server/Models/Asset.cs
﻿using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace ModelCabinet.Server.Models
{
    public class Asset
    {
        public int AssetId { get; set; }
        public string Name { get; set; }
        public string Path { get; set; }
        public DateTime DateCreation { get; set; }
        public DateTime DateUpdated { get; set; }
        public long FileSize { get; set; }
        [ForeignKey("Project")]
        public int ProjectId { get; set; }
        public virtual ICollection<AssetTag> AssetTags { get; set; }
    }
}

File: modelcabinet.client/src/app/Models/asset.ts
import { Tag } from "./tag";

export interface Asset {
  assetId: number,
  name: string,
  path: string,
  dateCreation: Date,
  dateUpdated: Date,
  fileSize: number, 
  projectId: number,
  assetTags: AssetTag[]
}

// Used to Match the structure in the backend
export interface AssetTag {
  assetId: number,
  tagId: number,
  asset: Asset,
  tag: Tag
}
